probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -27,6 +27,13 @@ pub trait ArmProbe {
     fn write_8(&mut self, ap: MemoryAp, address: u32, data: &[u8]) -> Result<(), Error>;
     fn write_32(&mut self, ap: MemoryAp, address: u32, data: &[u32]) -> Result<(), Error>;
 
+    /// Read bytes by the fastest means possible. May be implemented using 32
+    /// bit reads, so may read all 32 bit words that overlap with the requested
+    /// address range.
+    fn read(&mut self, ap: MemoryAp, address: u32, data: &mut [u8]) -> Result<(), Error> {
+        self.read_8(ap, address, data)
```

Comment:
I think this should be read_32. All chips support this and it's the fastest way. 8bit is not supported by all chips. In fact, only a few.

Reply:
Make sense. I've reimplemented it to use read_32 by default. I'm still working on testing my change, but thought I'd upload it straight away in case you have comments on the API.